target/linux/lantiq/patches-5.10/0801-remoteproc-add-avm_wasp-driver-kernel-module.patch

```diff
@@ -0,0 +1,1115 @@
++
++	  It's safe to say N here.
++
+ config IMX_REMOTEPROC
+ 	tristate "IMX6/7 remoteproc support"
+ 	depends on ARCH_MXC
+--- a/drivers/remoteproc/Makefile
++++ b/drivers/remoteproc/Makefile
+@@ -11,6 +11,7 @@ remoteproc-y				+= remoteproc_sysfs.o
+ remoteproc-y				+= remoteproc_virtio.o
+ remoteproc-y				+= remoteproc_elf_loader.o
+ obj-$(CONFIG_REMOTEPROC_CDEV)		+= remoteproc_cdev.o
++obj-$(CONFIG_AVM_WASP_REMOTEPROC)	+= avm_wasp.o
+ obj-$(CONFIG_IMX_REMOTEPROC)		+= imx_rproc.o
+ obj-$(CONFIG_INGENIC_VPU_RPROC)		+= ingenic_rproc.o
+ obj-$(CONFIG_MTK_SCP)			+= mtk_scp.o mtk_scp_ipi.o
+--- /dev/null
++++ b/drivers/remoteproc/avm_wasp.c
+@@ -0,0 +1,1051 @@
++// SPDX-License-Identifier: GPL-2.0-only
++/*
++ * AVM WASP Remote Processor driver
++ *
++ * Copyright (c) 2019-2020 Andreas Böhler
++ * Copyright (c) 2021-2022 Daniel Kestrel
++ *
++ */
++
++#include <linux/kernel.h>
++#include <linux/module.h>
++#include <linux/of_device.h>
++#include <linux/of_mdio.h>
++#include <linux/of_gpio.h>
++#include <linux/platform_device.h>
++#include <linux/remoteproc.h>
++#include <linux/timekeeping.h>
++#include <net/sock.h>
++#include <asm-generic/gpio.h>
++
++#include "remoteproc_internal.h"
++
++#define WASP_CHUNK_SIZE			14
++#define WASP_ADDR			0x07
++#define WASP_TIMEOUT_COUNT		1000
++#define WASP_WAIT_TIMEOUT_COUNT		20
++#define WASP_CHECK_LEN_DIVBY4_MASK	0x3
++
++#define WASP_WAIT_SLEEP_US_LOW		50000
++#define WASP_WAIT_SLEEP_US		100000
++#define WASP_POLL_SLEEP_US		200
++
++#define WASP_RESP_OK			0x0002
++#define WASP_RESP_READY_TO_START	0x0202
++
++#define WASP_CMD_SET_PARAMS		0x0c01
++#define WASP_CMD_SET_CHECKSUM_3390	0x0801
++#define WASP_CMD_SET_CHECKSUM_X490	0x0401
++#define WASP_CMD_SET_DATA		0x0e01
++#define WASP_CMD_START_FIRMWARE_3390	0x0201
++#define WASP_CMD_START_FIRMWARE_X490	0x0001
++#define WASP_CMD_START_FIRMWARE2_X490	0x0101
++
++#define ETH_TYPE_ATH_ECPS_FRAME		0x88bd
++#define ETH_BUF_SIZE			1056
++#define ETH_SEND_LOOP_TIMEOUT_SECS	60
++#define ETH_MAX_DATA_SIZE		1028
++#define ETH_DATA_SIZE			1024
++#define ETH_WASP_PACKET_ID		0x1200
++
++#define CMD_FIRMWARE_DATA		0x0104
++#define CMD_START_FIRMWARE		0xd400
++
++#define RESP_DISCOVER			0x0000
++#define RESP_OK				0x0100
++#define RESP_STARTING			0x0200
++#define RESP_ERROR			0x0300
++
++static const u32 m_load_addr = 0x81a00000;
++static const u32 m_start_and_exec_addr = 0xbd003000;
++
++static const char mac_data[WASP_CHUNK_SIZE] = {0xaa, 0xaa, 0xaa, 0xaa, 0xaa,
++		0xaa, 0x04, 0x20, 0x03, 0x00, 0x00, 0x00, 0x00, 0x00};
++
++static char wasp_mac[] = {0x00, 0xaa, 0xaa, 0xaa, 0xaa, 0xaa};
++
++enum {
++	MODEL_3390,
++	MODEL_X490,
++	MODEL_UNKNOWN
++} m_model = MODEL_UNKNOWN;
++
++struct wasp_packet {
++	struct ethhdr eh;
++	struct __packed {
++		u16	packet_start;
++		u8	pad_one[5];
++		u16	command;
++		u16	response;
++		u16	counter;
++		u8	pad_two;
++	} hdr;
++	u8	payload[ETH_MAX_DATA_SIZE];
++} __packed;
++
++static char *firmware = "ath9k-eeprom-ahb-18100000.wmac.bin";
```

Comment:
Please rename this to `firmware_name`. Same for the others. The name firmware confused me.